datajoint/condition.py

```diff
@@ -61,6 +63,18 @@ def append(self, restriction):
             super().append(restriction)
 
 
+@dataclass
+class Top:
+    """
+    A "restriction" to set the sorting clauses of a query. Since it is not a true
+    restriction, it has no effect on the WHERE clause.
+    """
+
+    limit: int = 10
+    order_by: Union[str, List[str]] = "KEY"
```

Comment:
I agree that we should not provide an option for `order_by=None`.  Restricting by `dj.Top` will always imply ordering. Ordering by primary key has no performance penalty.

Reply:
Since you are not allowing for `order_by=None`, then you should not be handling this case in the parsing code.

Reply:
Actually, the `QueryExpression.cursor` [method](https://github.com/A-Baji/datajoint-python/blob/top-restrcition/datajoint/expression.py#L653) can still pass in `None` arguments.